src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
-package core.basesyntax;
-
-public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
-    }
-}
+package core.basesyntax;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            int totalSalory = 0;
+            totalSalory = getTotalSalory(names[i],data,dateFrom,dateTo);
+            builder.append(System.lineSeparator()).append(names[i]);
+            builder.append(" - ").append(totalSalory);
+        }
+        return builder.toString();
+    }
+
+    public int getTotalSalory(String name, String[] data, String dateFrom, String dateTo) {
+        int totalSalary = 0;
+        for (int i = 0; i < data.length; i++) {
+            String[] temp = data[i].split(" ");
+            if (checkPeriod(temp[0], dateFrom, dateTo) & name.equals(temp[1])) {

```

Comment:
![Screenshot 2022-06-21 at 08 22 25](https://user-images.githubusercontent.com/60258373/174722152-db5ee9a5-9bdb-48b1-a23e-0ab1db125ae5.png)
Before you create PR with your homework, you need to go through the checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected